content/providers/05-observability/humanloop.mdx

```diff
@@ -0,0 +1,265 @@
+          dark
+        />
+      </Tab>
+    </Tabs>
+
+    Update your `.env` file to configure the OTLP Exporter:
+
+    ```bash filename=".env"
+    OTEL_EXPORTER_OTLP_ENDPOINT=https://api.humanloop.com/v5/import/otel
+    OTEL_EXPORTER_OTLP_PROTOCOL=http/json
+    OTEL_EXPORTER_OTLP_HEADERS="X-API-KEY=xxxxxx" # Humanloop API key
+    ```
+
+    Register the OpenTelemetry SDK and add Humanloop metadata to the spans. The `humanloopPromptPath` specifies the (Prompt File)[http://localhost:3001/docs/v5/explanation/prompts] in Humanloop to which the spans will be logged.
+
+    ```ts highlight="3-13,19"
+    import { openai } from '@ai-sdk/openai';
+    import { generateText } from 'ai';
+    import { NodeSDK } from '@opentelemetry/sdk-node';
+    import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
+    import dotenv from 'dotenv';
+
+    dotenv.config();
+
+    const sdk = new NodeSDK({
+      instrumentations: [getNodeAutoInstrumentations()],
+    });
+
+    sdk.start();
+
+    async function main() {
+      // ... Vercel AI SDK calls ...
+
+      // Must call shutdown to flush traces
+      await sdk.shutdown();
+    }
+
+    main().catch(console.error);
+    ```
+
+    Your calls to the AI SDK should now be logged to Humanloop.
+
+  </Tab>
+</Tabs>
+
+## Trace Grouping
+
+To group multiple AI SDK calls into a single Flow Log, create and pass a Flow Log ID to the telemetry metadata of each AI SDK call.
+
+1. Create a Flow Log in Humanloop
+2. Pass the Flow Log ID to each AI SDK call
+3. Update the Flow Log when all executions are complete
+
+The Flow Log serves as a parent container for all related Prompt Logs in Humanloop.
+
+```ts
+import { HumanloopClient } from 'humanloop';
+
+const humanloop = new HumanloopClient();
+
+async function main() {
+  const flow = await humanloop.flows.upsert({
+    path: 'Plethora of Poetry',
+    attributes: {},
+  });
+  const flowLog = await humanloop.flows.log({
+    id: flow.id,
+  });
+
+  const outputs = [];
+
+  for (const poetName of ['Edgar Allan Poe', 'Mary Shelley', 'Lord Byron']) {
+    const result = await generateText({
+      model: openai('gpt-3.5-turbo'),
+      maxTokens: 50,
+      prompt: `Write me a poem in the style of ${poetName}.`,
+      experimental_telemetry: {
+        isEnabled: true,
+        functionId: `poet-${poetName.toLowerCase().replace(' ', '-')}`,
+        metadata: {
+          humanloopFlowId: flowLog.id,
+          humanloopPromptPath: `Poets/${poetName}`,
+        },
+      },
+    });
+
+    outputs.push(result.text);
+  }
+
+  await humanloop.flows.updateLog(flowLog.id, {
+    traceStatus: 'complete',
+    output: outputs.join('\n\n'),
+  });
+
+  await sdk.shutdown();
+}
+```
+
+## Debugging
+
+If you aren't using Next.js 15+, you will also need to enable the experimental instrumentation hook (available in 13.4+).
```

Comment:
Need to elaborate this further: it's next.js only, should mention what the hook's benefits, THEN mentions you don't need extra configuration for >= 15